src/test/java/address/unittests/BrowserManagerTest.java

```diff
@@ -0,0 +1,76 @@
+package address.unittests;
+
+import address.browser.AddressBookBrowser;
+import address.browser.BrowserManager;
+import address.events.LocalModelChangedEvent;
+import address.model.ContactGroup;
+import address.model.Person;
+import address.util.JavafxThreadingRule;
+import address.util.TestUtil;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for the BrowserManager correct behaviours
+ */
+public class BrowserManagerTest {
+
+    @Rule
+    /**
+     * To run test cases on JavaFX thread.
+     */
+    public JavafxThreadingRule javaFxRule = new JavafxThreadingRule();
+
+    @Test
+    public void testHandleLocalModelChangedEvent_deletedPerson_personRemovedFromBrowserTab() {
+        List<Person> sampleList = TestUtil.generateSampleData();
+        ObservableList<Person> list = FXCollections.observableArrayList();
+        BrowserManager browserManager = new BrowserManager(list);
+        list.add(sampleList.get(0));
+        list.add(sampleList.get(1));
+        list.add(sampleList.get(2));
+        AddressBookBrowser browser = browserManager.getBrowser().get();
+
+        //Load index 0, 1, 2 in list
+        browser.loadProfilePage(list.get(0));
+
+        Person deletedPerson = list.remove(0);
+        browserManager.handleLocalModelChangedEvent(
+                new LocalModelChangedEvent(new ArrayList<Person>(), new ArrayList<ContactGroup>()));
+        //deleted person should be removed from the browser.
+        assertFalse(browser.getBrowserTabs().stream().filter(browserTab -> deletedPerson
+                                                     .equals(browserTab.getPerson())).findAny().isPresent());
+    }
+
+    @Test
+    public void testHandleLocalModelChangedEvent_editedPerson_personEditedInBrowserTab() {
+        List<Person> sampleList = TestUtil.generateSampleData();
+        ObservableList<Person> list = FXCollections.observableArrayList();
+        BrowserManager browserManager = new BrowserManager(list);
+        list.add(sampleList.get(0));
+        list.add(sampleList.get(1));
+        list.add(sampleList.get(2));
+        AddressBookBrowser browser = browserManager.getBrowser().get();
+
+        //Loads index 0, 1, 2 in list
+        browser.loadProfilePage(list.get(0));
+
+        //edit person in model
+        list.get(0).setGithubUserName("1");
+        browserManager.handleLocalModelChangedEvent(
+                new LocalModelChangedEvent(new ArrayList<Person>(), new ArrayList<ContactGroup>()));
+        //Check if person in the browser is edited.
+        assertTrue(browser.getBrowserTabs().stream().filter(browserTab
+                -> browserTab.getPerson() != null
+                   && browserTab.getPerson().getGithubUserName().equals("1")).findAny().isPresent());
```

Comment:
Should assert before changing that `list.get(0).getGithubUserName` is not `1`. If the sample data is ever changed such that the `githubUserName` is `1` even before changing, this test does not prove anything.